test/load/allocation/grpc/README.md

```diff
@@ -0,0 +1,107 @@
+
+## Kubernetes Cluster Setup
+
+For the load test you can follow the regular Kubernetes and Agones setup. In order to test the the allocation performance in isolation, we let the Agones to actuate all
+the game servers beefor starting the test.
+Here are the few important things:
+- Install Kubernetes in a region not zone to avoid being impacted by the rollouts
+- Consider using multiple cpu VMs for the agones-system node pool. We used n1-4 type VMs 
+- In the default-node pool (where the Game Server pods are created), make sure there are enough nodes available so Agones can actuate the game servers. We set the clustre count to 25 pre zonee (total of 75 per region)
+
+## Configuring the Allocator Service
+
+The allocator service uses gRPC. In order to be able to call the service, TLS and optionally mTLS has to be setup.
+For more information visit [agones-allocator]({{< relref "https://agones.dev/site/docs/advanced/allocator-service/">}}).
+
+## Fleet Setting
+
+We used the following fleet configuration during out testing. It creates 4000 simple-udp game servers. 
+Also uaing the automaticShutdownDelayMin parameter to 10, simple-udp game servers shutdown after 10 minutes.
+This helps to easily re-run the test without requiring to delete the game servers. 
+
+```yaml
+apiVersion: "agones.dev/v1"
+kind: Fleet
+metadata:
+  name: fleet-example
+spec:
+  # the number of GameServers to keep Ready or Allocated in this Fleet
+  replicas: 4000
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+  template:
+    metadata:
+      labels:
+        foo: bar
+    # GameServer specification
+    spec:
+        health: {}
+        ports:
+        - containerPort: 7654
+          name: default
+        sdkServer: {}
+        # The GameServer's Pod template
+        template:
+            spec:
+                containers:
+                - args:
+                  # We setup the simple-udp server to shutdown 10 mins after allocation 
+                  - -automaticShutdownDelayMin=10
+                  image: gcr.io/agones-images/udp-server:0.21
+                  name: simple-udp
+                  resources:
+                  limits:
+                    cpu: 20m
+                    memory: 32Mi
+                  requests:
+                    cpu: 20m
+                    memory: 32Mi
+```
+
+## Configuring the Allocator Service
+
+The allocator service uses gRPC. In order to be able to call the service, TLS and optionally mTLS has to be setup.
+For more information visit [agones-allocator]({{< relref "https://agones.dev/site/docs/advanced/allocator-service/">}}).
+
+## Running the test
+
+You can use the provided runAllocation.sh file. Before you run, you need to update the values in runAllocation.sh
+
+```
+TESTRUNSCOUNT=3
+NAMESPACE=default      # Namescape of the fleet
+EXTERNAL_IP=<IP_ADRESSS_TO_THE_ALLOCATOR_SERVICES_LOAD_BALANCER>
+KEY_FILE=client.key    # Path to mTLS client key file 
```

Comment:
Above it says that mTLS may be enabled. Are the KEY_FILE and CERT_FILE only required if it is?

Reply:
Reading the code these do not look optional.

Reply:
removed optional